splunk_connect_for_snmp/inventory/tasks.py

```diff
@@ -300,17 +300,20 @@ def inventory_setup_poller(self, work):
                             profile["condition"]["field"].replace(".", "|")
                         ]
 
-                        for pattern in profile["condition"]["patterns"]:
-
-                            result = re.match(pattern, cs["value"])
-                            if result:
-                                logger.debug(f"Adding smart profile {profile_name}")
-                                if profile["frequency"] not in assigned_profiles:
-                                    assigned_profiles[profile["frequency"]] = []
-                                assigned_profiles[profile["frequency"]].append(
-                                    profile_name
-                                )
-                                continue
+                        if type(profile["condition"]["patterns"]).__name__ != 'list':
```

Comment:
maybe would be better to use isinstance(profile["condition"]["patterns"], typing.List)?